### ISSUE TITLE
[Bug] Setting a left window margin clips the modeline on the right

### ISSUE THREAD

**Issue by Horrih** [seagle0128/doom-modeline#701]
### Thank you for the bug report

- [X] I am using the latest version of `doom-modeline` related packages.
- [X] I checked [FAQ](https://github.com/seagle0128/doom-modeline#faq).
- [X] You may also try reproduce the issue using clean environment and minimal configurations with
the command `emacs -Q`.

### Bug description

Hello,
I use margins to center the main window. This has the effect of clipping the modeline on the right.
**Substracting the left margin in _doom-modeline-def-modeline()_ seems to fix the issue**, however I am not confident enough to submit a PR, this function looks a bit tricky I'd mess it up.

Without margins :
![image](https://github.com/seagle0128/doom-modeline/assets/36566033/60faa1cf-6869-49b8-9104-9b645c975f9d)
With margins :
![image](https://github.com/seagle0128/doom-modeline/assets/36566033/2e5c18f8-608d-4bd9-a0c1-b609f7665619)

A partial work-around in the meantime is setting the following, but it does not look good :
```elisp
 (mode-line-right-align-edge 'right-margin)
```

### Steps to reproduce

I use emacs 29.2 in TUI, on RHEL 9, doom-modeline from MELPA (20240307.717)
Start emacs in TUI (emacs -nw) with the following init.el, then press Ctrl+r to apply a left margin

```elisp
(require 'package)
(package-initialize)
(use-package doom-modeline
  :init (doom-modeline-mode))

(defun my/add-left-margin()
  (interactive)
  (set-window-margins (selected-window) 20 0))

(keymap-global-set "C-r" #'my/add-left-margin)
```

### Expected behavior

Have the right modeline segments align to the right of the window instead of being clipped.

### OS

Linux

### Emacs Version

29 (gccemacs)

### Emacs Configurations

Vanilla + basic config

**Comment by seagle0128** [comment]
Please check `mode-line-right-align-edge` and try `right-margin`.

**Comment by Horrih** [comment]
> Please check `mode-line-right-align-edge` and try `right-margin`.

Thank you for your answer.
Setting this does take into account the right margin, but even then the results are weird.
I used the following minimalist init.el 
```elisp
(require 'package)
(package-initialize)
(use-package doom-modeline
  :custom (mode-line-right-align-edge 'window)
  :init (doom-modeline-mode))

(defun my/add-margins()
  (interactive)
  (set-window-margins (selected-window) 0 10))

(keymap-global-set "C-r" #'my/add-margins)
```
In the following screenshots I used the following margins : 0 on the left 10 on the right

With `(mode-line-right-align-edge 'window)`, it aligns with the right margin
![image](https://github.com/seagle0128/doom-modeline/assets/36566033/51860f17-457e-481e-9a21-876e805f6ada)

With `(mode-line-right-align-edge 'right-margin)`, it aligns with __twice__ the right margin
![image](https://github.com/seagle0128/doom-modeline/assets/36566033/15ec1585-10eb-4d22-8e3c-d580470f4fc2)

Now if I set the margins the other way : 10 on the left 0 on the right
With both settings, it aligns with the window on the left, gets clipped by 10 on the right
![image](https://github.com/seagle0128/doom-modeline/assets/36566033/1a473e6c-c9ff-4ff1-b876-1cb01c5bb52e)

Naively, I would expect the 'window setting to ignore margins for the modeline, and the 'right-margin setting to align the modeline with the margins on the right-side or both.
